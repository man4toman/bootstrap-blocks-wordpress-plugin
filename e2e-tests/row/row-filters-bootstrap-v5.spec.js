/**
 * WordPress dependencies
 */
import {
	activatePlugin,
	deactivatePlugin,
	createNewPost,
	getEditedPostContent,
} from '@wordpress/e2e-test-utils';
import { insertRowBlock, selectRowBlock } from './row-helper';
import {
	clickElementByText,
	getSelectedValueBySelectLabel,
	selectOption,
	selectOptionIsAvailable,
} from '../helper';

describe( 'row block filters Bootstrap 5', () => {
	beforeAll( async () => {
		await activatePlugin( 'wp-bootstrap-blocks-test-bootstrap-v5' );
		await activatePlugin( 'wp-bootstrap-blocks-test-row-filters' );
	} );

	afterAll( async () => {
		await deactivatePlugin( 'wp-bootstrap-blocks-test-bootstrap-v5' );
		await deactivatePlugin( 'wp-bootstrap-blocks-test-row-filters' );
	} );

	beforeEach( async () => {
		await createNewPost();
	} );

	it( 'wpBootstrapBlocks.row.horizontalGuttersOptions should add horizontal gutters option', async () => {
		await insertRowBlock();
		await selectRowBlock();

		// Disable No Gutters option to make Gutter options visible
		await clickElementByText( 'label', 'No Gutters' );

		// Additional horizontal gutters option should be available
		expect(
			await selectOptionIsAvailable( 'Horizontal Gutters', 'gx-4' )
		).toBe( true );

		// Horizontal Gutters option should be applied
		await selectOption( 'Horizontal Gutters', 'gx-4' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'wpBootstrapBlocks.row.verticalGuttersOptions should add vertical gutters option', async () => {
		await insertRowBlock();
		await selectRowBlock();

		// Disable No Gutters option to make Gutter options visible
		await clickElementByText( 'label', 'No Gutters' );

		// Additional vertical gutters option should be available
		expect(
			await selectOptionIsAvailable( 'Vertical Gutters', 'gy-4' )
		).toBe( true );

		// Vertical Gutters option should be applied
		await selectOption( 'Vertical Gutters', 'gy-4' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'wp_bootstrap_blocks_row_default_attributes should override default attributes with Bootstrap 5', async () => {
		await insertRowBlock();
		await selectRowBlock();

		// Disable No Gutters option to make Gutter options visible
		await clickElementByText( 'label', 'No Gutters' );

		// Horizontal Gutters should be selected
		expect(
			await getSelectedValueBySelectLabel( 'Horizontal Gutters' )
		).toMatch( 'gx-5' );

		// Vertical Gutters should be selected
		expect(
			await getSelectedValueBySelectLabel( 'Vertical Gutters' )
		).toMatch( 'gy-3' );

		// Check if attributes are set correctly
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
} );
