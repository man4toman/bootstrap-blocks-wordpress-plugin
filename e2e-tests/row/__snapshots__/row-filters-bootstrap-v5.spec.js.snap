// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`row block filters Bootstrap 5 wp_bootstrap_blocks_row_default_attributes should override default attributes with Bootstrap 5 1`] = `
"<!-- wp:wp-bootstrap-blocks/row {\\"noGutters\\":false} -->
<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":4} /-->

<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":8} /-->
<!-- /wp:wp-bootstrap-blocks/row -->"
`;

exports[`row block filters Bootstrap 5 wpBootstrapBlocks.row.horizontalGuttersOptions should add horizontal gutters option 1`] = `
"<!-- wp:wp-bootstrap-blocks/row {\\"noGutters\\":false,\\"horizontalGutters\\":\\"gx-4\\"} -->
<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":4} /-->

<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":8} /-->
<!-- /wp:wp-bootstrap-blocks/row -->"
`;

exports[`row block filters Bootstrap 5 wpBootstrapBlocks.row.verticalGuttersOptions should add vertical gutters option 1`] = `
"<!-- wp:wp-bootstrap-blocks/row {\\"noGutters\\":false,\\"verticalGutters\\":\\"gy-4\\"} -->
<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":4} /-->

<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":8} /-->
<!-- /wp:wp-bootstrap-blocks/row -->"
`;
